projects/plugins/jetpack/_inc/connect-button.js

```diff
@@ -25,33 +25,36 @@ jQuery( document ).ready( function ( $ ) {
 			connectionHelpSections.fadeOut( 600 );
 		}
 
-		jetpackConnectButton.selectAndStartConnectionFlow();
+		jetpackConnectButton.startConnectionFlow();
 	} );
 
 	var jetpackConnectButton = {
 		isRegistering: false,
 		isPaidPlan: false,
-		selectAndStartConnectionFlow: function () {
+		startConnectionFlow: function () {
 			var connectionHelpSections = $( '#jetpack-connection-cards, .jp-connect-full__testimonial' );
 			if ( connectionHelpSections.length ) {
 				connectionHelpSections.fadeOut( 600 );
 			}
 
 			if ( ! jetpackConnectButton.isRegistering ) {
-				if ( 'original' === jpConnect.forceVariation ) {
-					// Forcing original connection flow, `JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME = true`
-					// or we're dealing with Safari which has issues with handling 3rd party cookies.
-					jetpackConnectButton.handleOriginalFlow();
+				jetpackConnectButton.handleConnection();
+			}
+		},
+		selectAndStartAuthorizationFlow: function ( data ) {
+			if ( data.allowInplaceAuthorization && 'original' !== jpConnect.forceVariation ) {
+				jetpackConnectButton.handleAuthorizeInPlaceFlow( data );
+			} else {
+				// Forcing original connection flow, `JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME = true`
+				// or we're dealing with Safari which has issues with handling 3rd party cookies.
+				if ( data.alternateAuthorizeUrl ) {
+					window.location = data.alternateAuthorizeUrl;
 				} else {
-					// Default in-place connection flow.
-					jetpackConnectButton.handleConnectInPlaceFlow();
+					window.location = data.authorizeUrl;
 				}
 			}
 		},
-		handleOriginalFlow: function () {
```

Comment:
While we are removing this, `handleConnectionError` is still using it.
Note that `handleConnectionError` is used in two cases:

1. Error occurred on `POST /connection/register`
2. Browser does not support 3rd party cookies (See `wpcom_nocookie` case in `receiveData`)

Reply:
good catch, thanks